<a href="https://colab.research.google.com/github/ltoosaint24/DS-Unit-1-Sprint-2-Statistics/blob/master/Loveline_Toussaint__LS_DS_124_Introduction_to_Bayesian_Inference_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lambda School Data Science Module 124

## Introduction to Bayesian Inference




## Assignment - Code it up!

We used pure math to apply Bayes Theorem to drug tests. Now write Python code to reproduce the results! This is purposefully open ended - you'll have to think about how you should represent probabilities and events. You can and should look things up.

Specific goals/targets:

### 1) Write a function 

`def prob_drunk_given_positive(prob_drunk_prior, false_positive_rate, true_positive_rate):` 

You should only truly need these three values in order to apply Bayes Theorem. In this example, imagine that individuals are taking a breathalyzer test with an 8% false positive rate, a 100% true positive rate, and that our prior belief about drunk driving in the population is 1/1000. 
 - What is the probability that a person is drunk after one positive breathalyzer test?
 - What is the probability that a person is drunk after two positive breathalyzer tests?
 - How many positive breathalyzer tests are needed in order to have a probability that's greater than 95% that a person is drunk beyond the legal limit?


In [1]:
# TODO - code!
drinks_p = 1/1000
p_non_drinks = 1 - drinks_p
P_pos_drinks =0.08

numerator = p_non_drinks * drinks_p
marginal = (p_non_drinks * drinks_p) +(P_pos_drinks * drinks_p)
posterior_probability =numerator/marginal

print(posterior_probability)

0.9258572752548656


In [2]:
def prob_drink_use(drink_ps, false_positive, true_positive, number_of_test):
    posterior_probability = drink_ps
    for test in range(0, number_of_test):
      P_drinks = posterior_probability
      P_non_drink = 1-P_drinks
      P_pos_drinks = false_positive
      P_non_drinks = true_positive
      posterior_probability =(P_pos_drinks * P_drinks)/(P_pos_drinks*P_non_drinks) + (P_non_drinks * P_non_drinks)
      return posterior_probability

In [3]:
prob_drink_use(1/1000, 0.08, 1, 2)

1.001

Based on the  Bayes rule, it would take 2 test, because of the posterior probability value of 1.001.

### 2) In your own words, summarize the difference between Bayesian and Frequentist statistics

If you're unsure where to start, check out [this blog post of Bayes theorem with Python](https://dataconomy.com/2015/02/introduction-to-bayes-theorem-with-python/).

Bayesian and Frequentist statistics are methods that utilized the observation of data patterns to make predictive conclusions. This means the statistics provided enable observers to be able to predict the occurrence of data pattern, this also explains the distribution of results. Bayesian providence a statistical definition of confidence, while Frequentists pertains to analyzing frequency and data occurrence. This benefit within understanding the probability within outcomes, being that probability pertains to the pattern of data frequency, the statistics provides the ration that depicts the probability or ratio of a data sequence, or outcome.


### 3) Use the following Template to help come up with ideas for your Build Sprint Project: 



---

## Idea 1:

### You
What do you care about?

What do you know about?

What decisions do you face?




As a person, I tend to care about the environment within various domains, tiers, and installations. I care for the building of consciouness, and increase interactions that are often demonstrated within establishment of productivity. I know that knowledge is based on the constant, consistent, observation, and construction of environment. The ideal of philosophy, are about the ability to coordinate objects to also generate various objects of comfort, function, and most importantly life. The decisions that are often face would the within the how awareness can be obtain, and sequence of objects, along side of communication, that determines the built of conscioussness. This usually impact the time within understanding.



### Seven templates

In order to better understand the prompts here, please review the [Priceonomics Content Marketing Templates](https://priceonomics.com/introducing-priceonomics-content-marketing/)

Can you apply the templates to your topics?

**Geographic Variation:**

**Trend related to the news:**

**Who does that?:**

**Answering a question people care about:**

**Valuable to businesses:**

**What's the most popular?:**

**Cost/Money rankings:**


Geographic Variation:
The data observation would be based on the data observation of building capacities within various locations.
The data index would be based on the global count of spacous matter and occupancy.

occupancy_building = [['Canada':['New Foundland': ['Building 1': 234, 'Building 2': 345, 'Building 3': 565], '[Nova Scotia': ['Building 1': 335,'Building 2': 550 , 'Building 3': 660],'Tuktoyaktuk':['Building 1': 233, 'Building 2': 150, 'Building 3': 260]], 
 [[234,235,1234],[1500, 345,250],[125,345,450],
[350],[560],[670],
[250],[345],[450]],
[[560],[765],[850],
[1250],[540],[450]
[11250],[450],[550]]]

Trend related to news:
The related trend that would be related to the list of building occupancy will involve the any type of social patterns,event, and activities. The current reporting trend pertains to corona virus. This can potentially effects the building occupancy data ratio, because there may be less recorded building activities. However, the data set pertains the records of buildings and an index of the total occupants that can be within each building. 

Who does that:

The track records are based on the county, and city track logs. These logs involve the area maps that provides details of the land resources that are availiable and their features, such as building size, function, and occupancy.


Answering Questions People Care About:
These type questions usually involves the ability to build rapport. They are connecting questions that leads to obtaining various data frames. For example, there could be question pertaining to finding the nearest building, and rather the distance contribute to how many people can occupy the building at certain time.

Valuable Businesses:
The topic within the dataset provided are valuable to business because it deals with the commercial circulation of clients, spaces, and determines the production rates based on the the qauntity of customers that can be within a specific area.

What's the most popular:

The most popular would be the most frequent building occupancy ratio. So popularity would be similar to frequency, and it would an indication of building occupation ratio that is largely exercised within buildings.

Cost/Money ranking:
The cost and money value are significant within the occupancy ratio because its like infrastructure cost to have various calibrations.

### Misconceptions

What misconceptions do people have about your topic?

The misconceptions that people have about the current topic would be the time difference that may flunctuate the total occupants that may occupy a builiding within a point of time. For example, a building may occupy more people within a period of time, however, capacity pertains the ability to contain a certain amount of people without loosing function, comforts, and most importantly privacy. The misconceptions also be within the placement of objects that may affect the capacity ratio; for instances if there are furniture within the space that may contribute the number of people that can be within the area.


What data storytelling example inspires you?

Could you do a new hypothesis, for the same question?

Could you do a new question, for the same topic?

Could you do a new topic, with the same "style"?

The data storytelling example that inspires would be an example that inspires would be one that based on geographic variations. This present idea compares the variation within building capacity, this can significantly similar to the data storytelling example set of the would be from the UCI Machine Inventory: "Crop Mapping Using fused optical radar data set". The radar ratings and geographic variation within the crop areas being upserved is similar to observing room occupancy. The variable being observed are the number of occupants that are analagous to the number of crops capacity within an area.

https://archive.ics.uci.edu/ml/datasets/Crop+mapping+using+fused+optical-radar+data+set

```
`# This is formatted as code`
```



###Data

Where could you search for data about your topic?

I would have to conduct active surveys about various building, and their capacity ratios. There are engineering scholarly materials that are provided, along city maps that are provided current  records of building operations. Those records provide active records of the various building activities that include capacity within a rate of time.

---

## Idea 2:

### You
What do you care about?

What do you know about?

What decisions do you face?



Current cares consists of increase robotic and technological operations. I have increase my awareness within the various robotic operations that will be necessary to continue restoring, and constructing various activities. I currently have knowledge of various companies that are actively within a tertiary stage of humanoid operations, that contribute to the decisions that I would like to make. For example, the design of such mechanical operations would be necessary for the restoration of various environmental activities, and would be significantly necessary to further define technological order.

### Seven templates

In order to better understand the prompts here, please review the [Priceonomics Content Marketing Templates](https://priceonomics.com/introducing-priceonomics-content-marketing/)

Can you apply the templates to your topics?

**Geographic Variation:**

**Trend related to the news:**

**Who does that?:**

**Answering a question people care about:**

**Valuable to businesses:**

**What's the most popular?:**

**Cost/Money rankings:**

Geographic Variation: The variation would be within historical and monumental territories, and restoration needs. For example, there are areas within primary attributes of activites, compared to initial civilized areas of extreme conditions.

Trend related to the news: The relevant news would the current news report of expeditions, and current complete operations of Humanoid, and bot samples. These are signs of the current generational progress, and the markers that are within the identifying the requirements, pace, and attributes that benefit the process required for each geographic variation.

Who does that ?:
The topic pertains various engineering, social, and anthropoligist. It would require active, detail observation of sequence of mechanical concepts, along and the reconstruction of historical structures that have been improved into modern foundation.

Answering a questions people care about:

Most people would care about the time to complete the project, the type of installation, and price point, and concepts that pertain to the restoration. There will also be concern about about the options, and various domains that will be included within piecing the factors into a content place.

Valuable to businesses:
This is valuable to business because of the release, and depletion of stagnation. This also a valuable worthy operations that was meant to further dictate a progress timeline that contributes to functions of business.

What's the most popular?:
The most popular may within the ideals of realism, and significance. For example, the humanoid is based on perspective of life style, and significance would be within the area developmental concepts.

Cost/Money rankings:
The cost would be based on the both the acitivity, and rate of return of the output.

### Misconceptions

What misconceptions do people have about your topic?

The misconceptions within the topic would fall within cost of operations, and its connections to organizations. There may also be misconceptions within the type of organizations that can be practice to further experience the fall effect of the demands of the idea.

### Examples

What data storytelling example inspires you?

Could you do a new hypothesis, for the same question?

Could you do a new question, for the same topic?

Could you do a new topic, with the same "style"?

The data storytelling example that inspires this concept would be would be from the UCI Machine Inventory: Human Activity Recognition Using Smartphone Data Set. This data set was closely in relations to the intended topic of research regarding robotics, and humanoids.There can be an alternative hypothesis for the same question, however, the statement wouldn't pertain to the relationships between the data sets. The hypothesis could be based on the predictive statement regarding the various probability of outcomes. There can be new questions about the same topic, however the topic should remain constant within the same style.

###Data

Where could you search for data about your topic?


The data about robotic topic could be obtain from active production surveys. There may be active records during the construction, and operational procedures. These data sets would be most reliable and valid; however, they would also be restricted to the data group. 

## Resources

- [Worked example of Bayes rule calculation](https://en.wikipedia.org/wiki/Bayes'_theorem#Examples) (helpful as it fully breaks out the denominator)
- [Source code for mvsdist in scipy](https://github.com/scipy/scipy/blob/90534919e139d2a81c24bf08341734ff41a3db12/scipy/stats/morestats.py#L139)

## Stretch Goals:

- Go back and study the content from Modules 1 & 2 & 3 to make sure that you're really comfortable with them.
- Apply a Bayesian technique to a problem you previously worked (in an assignment or project work) on from a frequentist (standard) perspective
- Check out [PyMC3](https://docs.pymc.io/) (note this goes beyond hypothesis tests into modeling) - read the guides and work through some examples
- Take PyMC3 further - see if you can build something with it!